bitcoin/src/address.rs

```diff
@@ -1304,6 +1308,24 @@ mod tests {
             format!("Test {{ address: {} }}", addr_str));
     }
 
+    #[test]
+    fn test_require_network_error() {
+        // Tests the require_network's behaviour for erroneous network,
+        // but the main goal is to check if Error::NetworkValidation prints the address correctly
+
+        let addr_str = "32iVBEu4dxkUQk9dJbZUiBiQdmypcEyJRf";
+        let address: Address<NetworkUnchecked> = addr_str.parse().unwrap();
+        let result = address.require_network(Network::Testnet);
+
+        match result {
+            // result must be a NetworkValidation error
+            Err(e) => assert_eq!(
+                format!("{}", e),
+                format!("address {} belongs to network {} which is different from required {}", addr_str, Network::Bitcoin, Network::Testnet)),
```

Comment:
Testing strings like this isn't great and I'm not even sure if a test is needed for such straightforward code..

Reply:
well then reverting the last commit.

Reply:
I was just not sure if it would work since I could not run it and see the printed message :)